keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/nonprobabilistic/operators/multiple/ModelVertex.java

```diff
@@ -0,0 +1,45 @@
+package io.improbable.keanu.vertices.dbl.nonprobabilistic.operators.multiple;
+
+import io.improbable.keanu.tensor.Tensor;
+import io.improbable.keanu.tensor.bool.BooleanTensor;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.tensor.intgr.IntegerTensor;
+import io.improbable.keanu.vertices.NonProbabilistic;
+import io.improbable.keanu.vertices.Vertex;
+import io.improbable.keanu.vertices.VertexLabel;
+import io.improbable.keanu.vertices.bool.BoolVertex;
+import io.improbable.keanu.vertices.bool.nonprobabilistic.operators.BoolModelResultVertex;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.DoubleModelResultVertex;
+import io.improbable.keanu.vertices.intgr.IntegerVertex;
+import io.improbable.keanu.vertices.intgr.nonprobabilistic.operators.IntegerModelResultVertex;
+
+import java.util.Map;
+
+public interface ModelVertex<T> extends NonProbabilistic<T> {
+
+    void run(Map<VertexLabel, Vertex<? extends Tensor>> inputs);
```

Comment:
Do we need to re-specify the inputs here?  Shouldn't we just be able to get them from the input Vertices we've previously specified?

Reply:
Nice - deleted the argument!